design/ceph/object-store-realm.md

```diff
@@ -0,0 +1,253 @@
+
+Once the storage admin creates a realm and zonegroup(s) on a cluster, the admin is ready to add object-stores to the master zones in a realm. 
+The admin will specify his/her configuration settings in the `realm` section of the object-store like in the following `object-store.yaml`.
+
+All objectstores wanting to be in a master zone must have the same name as a `masterZone` in the objects store realm config.
+
+If the name of the objectstore does match a `masterZone` the object stores created will be added to the master zone.
+
+If the name of the objectstore does not match a `masterZone`, then a new zone will be created with the object stores created via the resource (see next section).
+
+This example config will add objectstores to the already created master zone `may-realm-zone-a`.
+These objectstores will go into the master zone, zonegroup, and realm created for the ObjectStoreRealm resource above.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStore
+metadata:
+  name: my-realm-zone-a
+  namespace: rook-ceph
+spec:
+  metadataPool:
+    failureDomain: host
+    replicated:
+      size: 3
+  dataPool:
+    failureDomain: device
+    erasureCoded:
+      dataChunks: 6
+      codingChunks: 2
+  gateway:
+    type: s3
+    port: 80
+    securePort: 443
+    instances: 3
+  realm:
+    name: my-realm
+    zonegroup: zonegroup-a
+```
+
+Now create the object store.
+```bash
+kubectl create -f object-store.yaml
+```
+
+### Creating a Non-Master Zone on the same cluster the object-store-realm was created.
+
+Once the storage admin creates a realm and zonegroup(s) and a master zone in a zonegroup, the admin can create non-master zones in the given zonegroup.
+
+To create a new zone, the `realm` section needs to be added to the object store configuration. The name of the objectstore cannot match a `masterZone` in the object store realm configuration.
+
+This example config will create an object store in a new zone, `my-realm-new-zone` in zonegroup `zonegroup-a` in realm `my-realm`.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStore
+metadata:
+  name: my-realm-new-zone
+  namespace: rook-ceph
+spec:
+  metadataPool:
+    failureDomain: host
+    replicated:
+      size: 3
+  dataPool:
+    failureDomain: device
+    erasureCoded:
+      dataChunks: 6
+      codingChunks: 2
+  gateway:
+    type: s3
+    port: 80
+    securePort: 443
+    instances: 3
+  realm:
+    name: my-realm
+    zonegroup: zonegroup-a
+```
+
+Now create the object store.
+```bash
+kubectl create -f object-store.yaml
+```
+#### Steps
+
+When the `realm` section is specified the Rook operator recognizes that the object store resource is part of a zone.
+
+1. If the name of the object store is not the name of a master zone, a Ceph zone is created as long as the zonegroup and realm have been created. Then the objectstores are added to the zone.
+2. If the name of the object store has the name of a master zone, the objectstores are added to the zone
```

Comment:
- What if I define a CephObjectStore for a secondary zone but it already exists within the realm?
- The verbiage around objectstores and zones is misleading. Can there be multiple objectstores to a zone? (I don't think so and hence the words "added" and plural objectstores is conflicting)